content/en/tracing/trace_collection/trace_context_propagation/_index.md

```diff
@@ -15,17 +15,616 @@ further_reading:
+    // - Key: "_datadog"
+    // - Value: MessageAttributeValue object
+    //   - DataType: "String"
+    //   - StringValue: <JSON map with key-value headers>
+    if (headers.TryGetValue("_datadog", out var messageAttributeValue)
+        && messageAttributeValue.StringValue is string jsonString)
+    {
+        var datadogDictionary = JsonConvert.DeserializeObject<Dictionary<string, string>>(jsonString);
+        if (datadogDictionary.TryGetValue(name, out string value))
+        {
+            return new[] { value };
+        }
+    }
+    return Enumerable.Empty<string>();
+}
+```
+
+When using the `SpanContextExtractor` API to trace Kafka consumer spans, set `DD_TRACE_KAFKA_CREATE_CONSUMER_SCOPE_ENABLED` to `false`. This ensures the consumer span is correctly closed immediately after the message is consumed from the topic, and the metadata (such as `partition` and `offset`) is recorded correctly. Spans created from Kafka messages using the `SpanContextExtractor` API are children of the producer span, and siblings of the consumer span.
+
+If you need to propagate trace context manually (for libraries that are not instrumented automatically, like the WCF client), you can use the `SpanContextInjection` API. Here is an example for WCF where `this` is the WCF client:
+
+```csharp
+
+using (OperationContextScope ocs = new OperationContextScope(this.InnerChannel))
+{
+  var spanContextInjector = new SpanContextInjector();
+  spanContextInjector.Inject(OperationContext.Current.OutgoingMessageHeaders, SetHeaderValues, Tracer.Instance.ActiveScope?.Span?.Context);
+}
+
+
+void SetHeaderValues(MessageHeaders headers, string name, string value)
+{
+    MessageHeader header = MessageHeader.CreateHeader(name, "datadog", value);
+    headers.Add(header);
+}
+```
+
+{{% /collapse-content %}}
+
+[1]: #datadog-format
+[2]: https://www.w3.org/TR/trace-context/
+[3]: https://github.com/openzipkin/b3-propagation#single-header
+[4]: https://github.com/openzipkin/b3-propagation#multiple-headers
+[5]: #none-format
+[6]: https://github.com/DataDog/dd-trace-dotnet/releases/tag/v2.48.0
+[7]: https://github.com/DataDog/dd-trace-dotnet/releases/tag/v2.22.0
+[8]: https://github.com/DataDog/dd-trace-dotnet/releases/tag/v2.42.0
+
+
+{{% /tab %}}
+
+{{< /tabs >}}
+
+## Custom header formats
+
+### Datadog format
+
+When the Datadog SDK is configured with the Datadog format for extraction or injection (possibly both), the Datadog SDK interacts with the following request headers:
+
+`x-datadog-trace-id`
+: Specifies the lower 64-bits of the 128-bit trace-id, in decimal format.
+
+`x-datadog-parent-id`
+: Specifies the 64-bits span-id of the current span, in decimal format.
+
+`x-datadog-origin`
+: Specifies the Datadog service that initiated the trace, such as [Real User Monitoring][7] or [Synthetic Monitoring][8]. If this header is present, the value is expected to be one of: `rum`, `synthetics`, `synthetics-browser`.
```

Comment:
@brett0000FF The use of the word `the Datadog service` doesn't feel right in this instance because the accepted values here are either `rum`, `synthetics`, `synthetics-browser`. Can we revert this change and use the word `product` or is there another term that you think would be more suitable?

Reply:
Gotcha, that works for me. I mainly changed this because one of our Vale rules checks for "Datadog product" and recommends "Datadog service" or "Datadog". In this context, I think it's fine to leave!